File: Java_Advanced/Testing_Assignment_5/src/main/java/com/vti/entity/Account.java
package com.vti.entity;

import java.io.Serializable;
import java.util.Date;
import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.Formula;

@Entity
@Table(name = "Account", catalog = "Testing_System_5")
@Inheritance(strategy = InheritanceType.JOINED)
public class Account implements Serializable {

	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	@Id
	@Column(name = "AccountID")
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private short id;

	@Column(name = "Email", length = 50, nullable = false, unique = true, updatable = false)
	private String email;

	@Column(name = "Username", length = 50, nullable = false, unique = true, updatable = false)
	private String userName;

	@Column(name = "FirstName", length = 50, nullable = false)
	private String firstName;

	@Column(name = "LastName", length = 50, nullable = false)
	private String lastName;

	@Formula("concat(FirstName, ' ' ,LastName)")
	private String fullName;

	@ManyToOne
	@JoinColumn(name = "DepartmentID", nullable = false)
	private Department department;

	@ManyToOne
	@JoinColumn(name = "PositionID", nullable = false)
	private Position position;

	@ManyToOne
	@JoinColumn(name = "SalaryID", nullable = false)
	private Salary salary;

	@Column(name = "CreateDate")
	@Temporal(TemporalType.TIMESTAMP)
	@CreationTimestamp
	private Date createDate;

	@OneToMany(mappedBy = "creator")
	private List<Group> createGroups;

	@OneToMany(mappedBy = "account")
	private List<GroupAccount> groups;

	public Account() {
		// TODO Auto-generated constructor stub
	}

	public short getId() {
		return id;
	}

	public void setId(short id) {
		this.id = id;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getUserName() {
		return userName;
	}

	public void setUserName(String userName) {
		this.userName = userName;
	}

	public String getFirstName() {
		return firstName;
	}

	public void setFirstName(String firstName) {
		this.firstName = firstName;
	}

	public String getLastName() {
		return lastName;
	}

	public void setLastName(String lastName) {
		this.lastName = lastName;
	}

	public String getFullName() {
		return fullName;
	}

	public void setFullName(String fullName) {
		this.fullName = fullName;
	}

	public Date getCreateDate() {
		return createDate;
	}

	public void setCreateDate(Date createDate) {
		this.createDate = createDate;
	}

	public Department getDepartment() {
		return department;
	}

	public void setDepartment(Department department) {
		this.department = department;
	}

	public Position getPosition() {
		return position;
	}

	public void setPosition(Position position) {
		this.position = position;
	}

	public Salary getSalary() {
		return salary;
	}

	public void setSalary(Salary salary) {
		this.salary = salary;
	}

	public List<Group> getCreateGroups() {
		return createGroups;
	}

	public void setCreateGroups(List<Group> createGroups) {
		this.createGroups = createGroups;
	}

	public List<GroupAccount> getGroups() {
		return groups;
	}

	public void setGroups(List<GroupAccount> groups) {
		this.groups = groups;
	}

	@Override
	public String toString() {
		return "Account [id=" + id + ", email=" + email + ", userName=" + userName + ", firstName=" + firstName
				+ ", lastName=" + lastName + ", fullName=" + fullName + ", department=" + department + ", position="
				+ position + ", salary=" + salary + ", createDate=" + createDate + "]";
	}

}

File: ReactJS/Lab 1/Lab 1.2_OOP/Account.ts
import { Department } from "./Department";
import { Possition } from "./Position";

class Account {
    // Khai báo các thuộc tính.
    private id: number;
    private email: string;
    private username: string;
    private fullname: string;
    private department: Department;
    private position: Possition;
    private createDate: Date;
    // Hàm tạo full tham số cho Account
    constructor(id: number, email: string, username: string, fullname: string, department: Department, position: Possition, createDate: Date) {
        this.id = id;
        this.email = email;
        this.username = username;
        this.fullname = fullname;
        this.department = department;
        this.position = position;
        this.createDate = createDate;

    }
    // Nhóm hàm getter và Setter
    public setid(v: number) {
        this.id = v;
    }
    public getid(): number {
        return this.id;
    }

    public setemail(v: string) {
        this.email = v;
    }
    public getemail(): string {
        return this.email;
    }

    public setusername(v: string) {
        this.username = v;
    }
    public getusername(): string {
        return this.username;
    }

    public setfullname(v: string) {
        this.fullname = v;
    }
    public getfullname(): string {
        return this.fullname;
    }

    public setdepartment(v: Department) {
        this.department = v;
    }
    public getdepartment(): Department {
        return this.department;
    }

    public setposition(v: Possition) {
        this.position = v;
    }
    public getposition(): Possition {
        return this.position;
    }

    public setcreateDate(v: Date) {
        this.createDate = v;
    }
    public getcreateDate(): Date {
        return this.createDate;
    }

}
// Hàm in thông tin Account, viết bên ngoài class
function printInforAccount(account: Account) {
    console.log("ID: " + account.getid() + " UserName: " + account.getusername() + " Fullname: " + account.getfullname() + " Department: " + account.getdepartment().getname() + " Position: " + account.getposition().getname());
}
// Export Account và hàm printInforAccount sang class khác để sử dụng.
export { Account, printInforAccount }